File: algorithms/implementation/python3/find_digits.py
#!/bin/python3

import sys


t = int(input().strip())
for a0 in range(t):
    n = int(input().strip())

    count = 0
    digits = str(n)
    for digit in digits:
        if int(digit) != 0:
            if n % int(digit) == 0:
                count += 1
    print(count)

File: algorithms/implementation/C#/FindDigits.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
class Solution {

    static void Main(String[] args) {
        int t = Convert.ToInt32(Console.ReadLine());
        for(int a0 = 0; a0 < t; a0++){
            int n = Convert.ToInt32(Console.ReadLine());
            
            var count = 0;
            var digits = n.ToString();
            foreach (var digit in digits)
            {
                if (Int32.Parse(digit.ToString()) != 0)
                {
                    if (n % Int32.Parse(digit.ToString()) == 0)
                    {
                        count += 1;   
                    }
                }
            }
            Console.WriteLine(count);
        }
    }
}
